core/src/test/java/org/mskcc/cbio/portal/util/TestRequestBodyGZipFilter.java

```diff
@@ -0,0 +1,174 @@
+
+    /**
+     * When I filter a request that doesn't have the gzip header,
+     * the request should not be wrapped, and chain.doFilter should
+     * be called on the same request object passed to the method.
+     */
+    @Test
+    public void testDoFilterEnabledNotGZipped() throws Exception {
+        RequestBodyGZipFilter subject = createRequestBodyGZipFilter(true, 50000000);
+
+        HttpServletRequest request = createHttpServletRequest(null, "POST");
+        HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+        FilterChain chain = Mockito.mock(FilterChain.class);
+
+        subject.doFilter(request, response, chain);
+        
+        // request should not be wrapped
+        Mockito.verify(chain, Mockito.times(1)).doFilter(request, response);
+    }
+
+    /**
+     * When I filter a request that has the gzip header, but the gzip
+     * feature is disabled, the doFilter method should throw an exception.
+     */
+    @Test(expected = IllegalStateException.class)
+    public void testDoFilterDisabledGZipped() throws Exception {
+        RequestBodyGZipFilter subject = createRequestBodyGZipFilter(false, 50000000);
+
+        HttpServletRequest request = createHttpServletRequest("gzip", "POST");
+        HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+        FilterChain chain = Mockito.mock(FilterChain.class);
+
+        subject.doFilter(request, response, chain);
+    }
+
+    /**
+     * When I filter a request that has the gzip header and the gzip
+     * feature is enabled, but the method is something other than POST,
+     * the doFilter method should throw an exception.
+     */
+    @Test(expected = IllegalStateException.class)
+    public void testDoFilterEnabledGZippedBadMethod() throws Exception {
+        RequestBodyGZipFilter subject = createRequestBodyGZipFilter(true, 50000000);
+
+        HttpServletRequest request = createHttpServletRequest("gzip", "GET");
+        HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+        FilterChain chain = Mockito.mock(FilterChain.class);
+
+        subject.doFilter(request, response, chain);
+    }
+
+    /**
+     * When I filter a request that has the gzip header and the gzip
+     * feature is enabled and the method is POST, the doFilter method
+     * should wrap the request with a GzippedInputStreamWrapper.
+     * The BufferedReader for this GzippedInputStreamWrapper should return
+     * the contents of the gzipped file used to create the original input stream.
+     */
+    @Test
+    public void testDoFilterEnabledGZipped() throws Exception {
+        RequestBodyGZipFilter subject = createRequestBodyGZipFilter(true, 50000000);
+
+        // Not thrilled to be using the file system, but this was the most readable way to do this
+        // that I could think of.
+        // gzip_filter.json.gz is a gzip of a json file that contains the following: {"field":"value"}
```

Comment:
Interested if anyone has an idea on how to do this better.

Reply:
The total file size is 55 bytes .. perhaps it is reasonable to encode it directly as a byte array in code using escaped byte literals? The file is not human readable in any case (although I guess as a file it can be unzipped at the command line to "look inside" .. but the comment above handles that in the code.

Reply:
I like it. Thanks for the suggestion.